app/assets/javascripts/modules/accordion-with-descriptions.js

```diff
@@ -148,33 +168,39 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
           var action = '';
 
           // update button text
-          if ($openOrCloseAllButton.text() == "Expand all") {
+          if ($openOrCloseAllButton.text() == "Open all") {
             $openOrCloseAllButton.text("Close all");
             $openOrCloseAllButton.attr("aria-expanded", "true");
             action = 'open';
+
+            track('pageElementInteraction', 'accordionAllOpened', {
+              label: 'Open All'
+            });
           } else {
-            $openOrCloseAllButton.text("Expand all");
+            $openOrCloseAllButton.text("Open all");
             $openOrCloseAllButton.attr("aria-expanded", "false");
             action = 'close';
+
+            track('pageElementInteraction', 'accordionAllClosed', {
+              label: 'Close All'
+            });
           }
 
-          // Set aria-expanded for each button
-          $subsectionButton.each(function( index ) {
-            if (action == 'open') {
-              setExpandedState($(this), "true");
-            } else {
-              setExpandedState($(this), "false");
-            }
-          });
+          $element.find('.js-subsection').each(function() {
+            var $subsection = $(this);
+            var $button = $subsection.find('.js-subsection-button');
+            var $subsectionContent = $subsection.find('.js-subsection-content');
 
-          // show/hide content
-          $subsectionHeader.each(function( index ) {
             if (action == 'open') {
-              openSection($(this).next());
-              showOpenIcon($(this));
+              $button.attr("aria-expanded", "true");
+              $subsectionContent.removeClass('js-hidden');
```

Comment:
(I personally think these add/remove classes should have been combined into a [`.toggleClass()`](http://api.jquery.com/toggleclass/), but I realise this is just a direct read-across so fine to leave)